tools/gnur-runtime/src/main/java/org/renjin/gnur/api/Rinternals.java

```diff
@@ -1277,7 +1278,7 @@ public static void Rf_defineVar(SEXP nameSexp, SEXP value, SEXP rhoSexp) {
     Symbol name = (Symbol) nameSexp;
     Environment rho = (Environment) rhoSexp;
     
-    rho.setVariable(name, value);
+    rho.setVariable(Context.newTopLevelContext(), name, value);
```

Comment:
There should be **NO** calls to `Context.newTopLevelContext()`: this creates a new Renjin Session. Use `Native.currentContext()` like other methods in this class that require a `Context` reference.